### DISCUSSION TITLE
Disabling CSRF in Laravel 11

### DISCUSSION THREAD

**Discussion by colinmackinlay** [spatie/laravel-support-bubble#50] [Bugs] [answered]
Since upgrading to L11 I get an error when submitting support bubble requests: `CSRF token mismatch.`

I have the suggested exception in`VerifyCsrfToken.php`:
```
<?php

namespace App\Http\Middleware;

use Illuminate\Foundation\Http\Middleware\VerifyCsrfToken as Middleware;

class VerifyCsrfToken extends Middleware
{
    protected $except = [
        'support-bubble',
    ];
}
```
This worked fine pre-upgrade but now that middleware is dealt with differently this class isn't part of the applied middleware. I don't really want to manually apply all the standard middleware and just change VerifyCsrfToken to my own one.

Can anyone help?

PS I've filed this under bugs but its probably more about a documentation update being needed for L11 and CSRF

**Comment by jonodonovan** [comment]
Did you get it to work?

**Reply by colinmackinlay** [reply]
@jonodonovan unfortunately not. I've disabled it entirely for now which is a shame. If I get time I'd like to try it in a fresh L11 installation which I expect to work and then try to work out what breaks it in my setup.

**Reply by jonodonovan** [reply, marked as answer]
I found the fix

In boostrap/app add

```
->withMiddleware(function (Middleware $middleware) {
        $middleware->validateCsrfTokens(except: [
			'support-bubble',
		]);
    })
```

Then update the env file from

`QUEUE_CONNECTION=database`

to 

`QUEUE_CONNECTION=sync`

Tested working locally and prod

**Reply by colinmackinlay** [reply]
@jonodonovan magic! Fixed for me - didn't need to change my queue in local or production. I've created a pull request to `readme.md`
